lms/static/js/edxnotes/plugins/llm_summarize.js

```diff
@@ -68,17 +68,113 @@
+                    // inside the annotatorjs modal.
+                    .loader {
+                        width: 5em !important;
+                        border-color: red !important;
+                    }
+                    @keyframes rotation {
+                    0% {
+                        transform: rotate(0deg);
+                    }
+                    100% {
+                        transform: rotate(360deg);
+                    }
                 `;
                 document.head.appendChild(style);
-                var annotator = this.annotator;
+                this.modifyDom(this.annotator);
+                const summarizeButton = document.getElementById('summarizeButton');
+                summarizeButton.addEventListener('click', function(ev) {
+                    document.querySelector('.annotator-outer.annotator-editor').setAttribute('is_summarizing', true);
+                });
+                this.annotator.subscribe('annotationEditorShown', this.handleSummarize);
+                this.annotator.subscribe('annotationEditorHidden', this.cleanupSummarize);
+            },
+            handleSummarize: function (editor, annotation) {
+                if (editor.element[0].getAttribute('is_summarizing') !== 'true') return;
+                const textAreaWrapper = editor.fields[0].element;
+                editor.fields[1].element.children[0].value = 'ai_summary';
+                toggleLoader(editor);
+
+                const request = new Request("/pearson-core/llm-assistance/api/v0/summarize-text", {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json',
+                        'X-CSRFToken': $.cookie('csrftoken')
+                    },
+                    body: JSON.stringify({
+                        text_to_summarize: annotation.quote
+                    })
+                });
+                fetch(request)
+                    .then((response) => {
+                        if (response.ok) return response.json();
+                        throw new Error(gettext('There was an error while summarizing the content.'));
+                    })
+                    .then((data) => {
+                        toggleLoader();
+                        textAreaWrapper.children[0].value = data.summary;
+                    })
+                    .catch((error) => {
+                        alert(error.message);
+                        editor.hide();
+                    });
+
+                function toggleLoader() {
+                    const saveButton = document.querySelector('.annotator-controls .annotator-save');
+                    const loaderWrapper = document.querySelector('.summarize-loader-wrapper');
+                    const textArea = editor.fields[0].element.children[0];
+                    if (textArea.style.display === 'none') textArea.style.display = 'block';
+                    else textArea.style.display = 'none';
```

Comment:
Javascript has the toggleAttribute: https://developer.mozilla.org/en-US/docs/Web/API/Element/toggleAttribute ;jQuery has a .toggle function to handle this cases or you can use ternary operator to add the block or none value to the display attribute. @alexjmpb